implementation/src/main/java/io/smallrye/mutiny/operators/multi/MultiOnSubscribeCall.java

```diff
@@ -63,6 +66,61 @@ public void onSubscribe(Flow.Subscription s) {
                 s.cancel();
             }
         }
+
+        @Override
+        public void onFailure(Throwable throwable) {
+            lock.lock();
+            if (!uniHasTerminated) {
+                terminatedEarly = true;
+                this.failure = throwable;
+                lock.unlock();
+            } else {
+                lock.unlock();
+                super.onFailure(throwable);
+            }
+        }
+
+        @Override
+        public void onCompletion() {
+            lock.lock();
+            if (!uniHasTerminated) {
+                terminatedEarly = true;
+                lock.unlock();
+            } else {
+                lock.unlock();
+                super.onCompletion();
+            }
+        }
+
+        private void uniFailed(Throwable failure) {
+            getAndSetUpstreamSubscription(CANCELLED).cancel();
+            lock.lock();
+            uniHasTerminated = true;
```

Comment:
Here a try/finally could work. But except adding the suppressed, not sure it worths it.

Reply:
I tend not to wrap in `try` blocks if not doing I/O 😉 

Adding the suppressed exception is better I think, else we know it's swallowed.